CRM/Utils/Check/Component/Case.php

```diff
@@ -162,4 +162,239 @@ public function checkNullTimestamps() {
     return $messages;
   }
 
+  /**
+   * Check that the relationship types aren't going to cause problems.
+   *
+   * @return array<CRM_Utils_Check_Message>
+   *   An empty array, or a list of warnings
+   */
+  public function checkRelationshipTypeProblems() {
```

Comment:
This is a Status Check that no user can resolve without direct database access. There is no capability to edit the Relationship Type, Name in the user interface. So how was this envisioned to be actioned?

Reply:
Thanks for the comments.

The reason I commented here was because we had a client that has been using CiviCase for just over 5 months now. They set up the Relationship Types themselves for the Cases and in doing so, had created Relationship Types which had the same Label and same Name in both directions for the each relationship. The terminology used is generic so made sense for them to do so.

Case workflow was then set up and cases created.

Then this Status Check appears and part of the issue can be resolved. The Labels can be changed. However, changing the Name cannot be resolved without direct database access. And in doing that change to satisfy the Status Check criteria, then the Case Workflow is clobbered and needs to be set up again.

Reply:
I'm maybe missing a piece of info. E.g. "Spouse of" is a legitimate type where label/name are the same and the same in both direction. That doesn't create a status check warning.

I do understand that the UI can allow complex "I changed my mind" manipulations that end up in a problematic config, e.g. https://docs.civicrm.org/user/en/latest/case-management/what-you-need-to-know/#relationship-type-cross-duplication. I'd support enhancements that prevent this at configuration time.

Reply:
Thanks @demeritcowboy 

I do think there is a need for UI (& API?) validation to prevent ambiguous Relationship Type, _Labels_ being created in the first place.

I do not know why the Relationship Type, _Name_ is not already a randomly assigned unique string - because if it is truly used for "machine only" purposes and not editable in the UI then why does it need to be human readable at all? Perhaps there is some history here as to why it is what it is today, but still causing problems.

Reply:
> I do not know why the Relationship Type, Name is not already a randomly assigned unique string

Yes yes yes. Also change "name" to "machineName" or "internalLookupKey" or something that tells people it's not for humans and is more like a number, even though it can be a string if desired when created by extensions. I've been advocating for this for a while but it's a project.

> Perhaps there is some history

Yes. Label never used to exist. Name was effectively label and there was no label field. The distinction was introduced to allow for portable code between installs and translation, so that "id" (aka option_value.value, the stored value) could change between installs but code could reference, e.g. "Meeting", and that would work on any install regardless of the stored value and the translated label. It was decided to just leave existing name fields as it they were, with it's labelly words, rather than changing it to something more computery, and it stuck.

Also, don't let users do stuff. That usually works too.